### PR TITLE
Added rev2 PCB and moved original PCB to rev1

### PR DESCRIPTION
**Pull request by trinity1561**
New PCB was provided by Exclusive after group buy for E6.5 that was different than the one that came with the group buy.  Created new firmware for rev2 and moved the original firmware into rev1 folder.

Moved original firmware into e65/rev1 and added new firmware under e65/rev2.

## Description

Moved original e65 files into rev1 and created new rev2 folder for the new PCB.  All files e65.h and e65.c got renamed to rev1.h and rev1.c.  Updated readme.md files to named changes.

Created rev2 firmware including VIA keymap.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Issues Fixed or Closed by This PR

* Added rev2 PCB support

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by Wilba6582** [comment]
I suggest updating the rev1 vendor/product IDs at this time (change it from 0xFEED)

**Comment by vomindoraan** [comment]
Does rev2 support the same physical layouts as rev1?

**Comment by trinity1561** [comment]
> I suggest updating the rev1 vendor/product IDs at this time (change it from 0xFEED)

Doh, I was going to do this but completely forgot about it.  I'll try to add that change.

> Does rev2 support the same physical layouts as rev1?

Yes it does.  If all those layouts need to be added before being approved, I'll go ahead and add them.  I just wanted to get the VIA support out so people can use it (which has those supported layouts).

**Comment by vomindoraan** [comment]
In that case, `keymaps` should be kept in the base keyboard folder instead of being moved into `rev1`. It would also be good to make sure that the layout names match the existing ones, e.g. `LAYOUT_ANSI_BLOCKER_ALL` should be `LAYOUT_65_all_blocker` (I think).

**Comment by trinity1561** [comment]
> In that case, `keymaps` should be kept in the base keyboard folder instead of being moved into `rev1`. It would also be good to make sure that the layout names match the existing ones, e.g. `LAYOUT_ANSI_BLOCKER_ALL` should be `LAYOUT_65_all_blocker` (I think).

OK that makes sense.  Would it be better if I close this PR and make another one since there'll be alot of changes?  Or just update this one?

**Comment by vomindoraan** [comment]
Updating this PR would be better.

File: keyboards/exclusive/e65/rev2/rev2.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    // put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
+    // put your keyboard LED indicator (ex: Caps Lock LED) toggling code here
+    if (IS_LED_ON(usb_led, USB_LED_CAPS_LOCK)) {
+      writePinLow(C7);
+    } else {
+      writePinHigh(C7);
+    }
+
+    led_set_user(usb_led);
+}
```
Comment (drashna) [inline keyboards/exclusive/e65/rev2/rev2.c:50]: Should be replaced with: 
```suggestion
bool led_update_kb(led_t led_state) {
    bool res = led_update_user(led_state);
    if(res) {
        // writePin sets the pin high for 1 and low for 0.
        // In this example the pins are inverted, setting
        // it low/0 turns it on, and high/1 turns the LED off.
        // This behavior depends on whether the LED is between the pin
        // and VCC or the pin and GND.
        writePin(C7, !led_state.caps_lock);
    }
    return res;
}
```

File: keyboards/exclusive/e65/rev2/rev2.h [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "quantum.h"
+
+// layout with all the switches supported by the PCB
+#define LAYOUT_ALL( \
```
Comment (drashna) [inline keyboards/exclusive/e65/rev2/rev2.h:21]: These should be lowercase, except for the `LAYOUT` part. 
```suggestion
#define LAYOUT_all( \
```

File: keyboards/exclusive/e65/rev2/rev2.h [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+) { \
+    { K00,  K01,   K02,   K03,   K04,   K05,   K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D,   K0E   }, \
+    { K10,  K11,   K12,   K13,   K14,   K15,   K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D,   K1E   }, \
+    { K20,  K21,   K22,   K23,   K24,   K25,   K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   KC_NO, K2E   }, \
+    { K30,  K31,   K32,   K33,   K34,   K35,   K36,   K37,   K38,   K39,   K3A,   K3B,   K3C,   K3D,   K3E   }, \
+    { K40,  K41,   K42,   KC_NO, KC_NO, KC_NO, K46,   KC_NO, K48,   K49,   K4A,   K4B,   K4C,   K4D,   K4E   }, \
+}
+
+// ANSI layout
+#define LAYOUT_ANSI_BLOCKER_ALL( \
```
Comment (drashna) [inline keyboards/exclusive/e65/rev2/rev2.h:36]:
```suggestion
#define LAYOUT_ansi_blocker_all( \
```

File: keyboards/exclusive/e65/rev2/rev2.h [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    { K00,   K01,   K02,   K03,   K04,   K05,     K06,   K07,   K08,   K09,   K0A,   K0B,   K0C,   K0D,   K0E   }, \
+    { K10,   K11,   K12,   K13,   K14,   K15,     K16,   K17,   K18,   K19,   K1A,   K1B,   K1C,   K1D, K1E   }, \
+    { K20,   K21,   K22,   K23,   K24,   K25,     K26,   K27,   K28,   K29,   K2A,   K2B,   K2C,   KC_NO, K2E   }, \
+    { K30,   K31,   K32,   K33,   K34,   K35,     K36,   K37,   K38,   K39,   K3A,   K3B,   K3C,   K3D,   K3E   }, \
+    { K40,   K41,   K42,   KC_NO, KC_NO,  KC_NO,  K46,   KC_NO, K48,   K49,   KC_NO, K4B,   K4C,   K4D,   K4E   }, \
+}
+
+
+// generated by KBFirmware JSON to QMK Parser
+// https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (drashna) [inline keyboards/exclusive/e65/rev2/rev2.h:67]:
```suggestion
```

File: keyboards/exclusive/e65/readme.md [hunk 5]
```diff
@@ -1,15 +1,19 @@
...
-Keyboard Maintainer: [masterzen](https://github.com/masterzen)  
-Hardware Supported: E6.5 QMK PCB LED  
-Hardware Availability: [https://geekhack.org/index.php?topic=99655.msg2730615#msg2730615](https://geekhack.org/index.php?topic=99655.msg2730615#msg2730615)
 
-Make example for this keyboard (after setting up your build environment):
+ **Firmware files are SPECIFIC to each board. Firmware files from one, will not work on the other.** Please use the `.hex` appropriate for your board.
 
-    make exclusive/e65:default
+* Keyboard Maintainer for [Rev1](rev/): [masterzen](https://github.com/masterzen)  
+* Keyboard Maintainer fore [Rev2](rev2/): [Trinity1561](https://github.com/trinit1561)
```
Comment (tzarc) [inline keyboards/exclusive/e65/readme.md:12]:
```suggestion
* Keyboard Maintainer for [Rev2](rev2/): [Trinity1561](https://github.com/trinit1561)
```

File: keyboards/exclusive/e65/rev2/config.h [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x4558    //set to EX
+#define PRODUCT_ID      0x3635    //set to 65
+#define DEVICE_VER      0x0002    //version 2
+#define MANUFACTURER    Exclusive
+#define PRODUCT         E6.5 rev 2
+#define DESCRIPTION     This is the Black PCB for E6.5
```
Comment (tzarc) [inline keyboards/exclusive/e65/rev2/config.h:28]:
```suggestion
```

File: keyboards/exclusive/e65/rev2/config.h [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+//     #define RGBLIGHT_EFFECT_SNAKE
+//     #define RGBLIGHT_EFFECT_KNIGHT
+//     #define RGBLIGHT_EFFECT_CHRISTMAS
+//     #define RGBLIGHT_EFFECT_STATIC_GRADIENT
+//     #define RGBLIGHT_EFFECT_RGB_TEST
+//     #define RGBLIGHT_EFFECT_ALTERNATING
+#endif
+
+// generated by KBFirmware JSON to QMK Parser
+// https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (tzarc) [inline keyboards/exclusive/e65/rev2/config.h:76]:
```suggestion
```

File: keyboards/exclusive/e65/rev2/rev2.c [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "rev2.h"
+
+void matrix_init_kb(void) {
+    // put your keyboard start-up code here
+    // runs once when the firmware starts up
+
```
Comment (tzarc) [inline keyboards/exclusive/e65/rev2/rev2.c:21]:
```suggestion
```

File: keyboards/exclusive/e65/rev2/rev2.c [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    matrix_scan_user();
+}
+
+bool process_record_kb(uint16_t keycode, keyrecord_t *record) {
+    // put your per-action keyboard code here
+    // runs for every action, just before processing by the firmware
+
+    return process_record_user(keycode, record);
+}
+
```
Comment (tzarc) [inline keyboards/exclusive/e65/rev2/rev2.c:39]:
```suggestion
```

These are implemented in core already.

File: keyboards/exclusive/e65/rev2/rev2.c [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+    } else {
+      writePinHigh(C7);
+    }
+
+    led_set_user(usb_led);
+}
+
+
+// generated by KBFirmware JSON to QMK Parser
+// https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (tzarc) [inline keyboards/exclusive/e65/rev2/rev2.c:54]:
```suggestion
```

File: keyboards/exclusive/e65/rev2/rules.mk [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+MCU = atmega32u4
+
+# Bootloader selection
+#   Teensy       halfkay
+#   Pro Micro    caterina
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
```
Comment (tzarc) [inline keyboards/exclusive/e65/rev2/rules.mk:11]:
```suggestion
```

File: keyboards/exclusive/e65/rev2/rules.mk [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+EXTRAKEY_ENABLE = yes        # Audio control and System control
+CONSOLE_ENABLE = yes         # Console for debug
+COMMAND_ENABLE = yes         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no        # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no             # USB Nkey Rollover
+BACKLIGHT_ENABLE = yes       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes        # Enable keyboard RGB underglow
+MIDI_ENABLE = no             # MIDI support
```
Comment (tzarc) [inline keyboards/exclusive/e65/rev2/rules.mk:28]:
```suggestion
```

File: keyboards/exclusive/e65/rev2/rules.mk [hunk 13]
```diff
@@ -0,0 +1,36 @@
...
+UNICODE_ENABLE = no          # Unicode
+BLUETOOTH_ENABLE = no        # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no            # Audio output on port C6
+FAUXCLICKY_ENABLE = no       # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no          # Enable support for HD44780 based LCDs
+
+# generated by KBFirmware JSON to QMK Parser
+# https://noroadsleft.github.io/kbf_qmk_converter/
```
Comment (tzarc) [inline keyboards/exclusive/e65/rev2/rules.mk:36]:
```suggestion
```

**Pull request closed without merging**